index.html

```diff
@@ -889,67 +889,133 @@ <h3>Supporting algorithms</h3>
+              initiators of active Picture-in-Picture sessions</a>:
+            <ol>
+              <li>
+                If |relevantGlobal|'s [=relevant settings object=]'s [=origin=] is [=same origin=] with |origin|, return true.
+              </li>
+            </ol>
+          </li>
+          <li>
+            If |relevantGlobal|'s [=browsing context=] is [=context is capturing|capturing=], return true.
+          </li>
+          <li>
+            Let |topLevelBC| be |relevantGlobal|'s [=browsing context=]'s [=top-level browsing context=].
+          </li>
+          <li>
+            If |topLevelBC| does not have [=top-level traversable/system focus=], return false.
+          </li>
+          <li>
+            Let |focusedDocument| be the |topLevelBC|'s
+            <a href="https://html.spec.whatwg.org/multipage/interaction.html#currently-focused-area-of-a-top-level-browsing-context">
+              currently focused area</a>'s [=Node/node document=].
+          </li>
+          <li>
+            If |relevantGlobal|'s [=relevant settings object=]'s [=origin=] is [=same origin=] with
+            |focusedDocument|'s [=origin=], return true.
+          </li>
+          <li>
+            Otherwise, return false.
+          </li>
+        </ol>
+      </p>
+      <p>
+        The <dfn>passes privacy test</dfn> steps given the argument |observer:PressureObserver| and
+        its [=relevant global object=] |relevantGlobal|, are as follows:
+        <ol>
+          <li>
+            If |relevantGlobal| is a {{WorkerGlobalScope}} object:
+            <ol>
+              <li>
+                If |relevantGlobal|'s relevant worker is not a <a href="https://html.spec.whatwg.org/multipage/workers.html#active-needed-worker">
+                active needed worker</a>, return false.
+              </li>
+              <li>
+                If |relevantGlobal| is a {{DedicatedWorkerGlobalScope}} object:
+                <ol>
+                  <li>
+                    Let |ownerGlobal| be |relevantGlobal|'s [=owning global object=].
+                  </li>
+                  <li>
+                    Assert: |ownerGlobal| is not null.
+                  </li>
+                  <li>
+                    Return the result of running [=passes window privacy test=] with |observer| and |ownerGlobal|.
+                  </li>
+                </ol>
+              </li>
+              <li>
+                If |relevantGlobal| is a {{SharedWorkerGlobalScope}} object:
+                <ol>
+                  <li>
+                    [=list/For each=] |owner| in |relevantGlobal|'s [=WorkerGlobalScope/owner set=]:
+                    <ol>
+                      <li>
+                        Assert: |owner| is {{Document}}.
+                      </li>
+                      <li>
+                        If the result of running [=passes window privacy test=] with |observer| and |owner|'s
```

Comment:
This algorithm is returning true if "passes window privacy test" is true for any owner, so I'd like to confirm if this is indeed the intention.

This means that if I have, for instance:
* Window W1 showing https://foo.com/a.html that spawns a shared worker
* Window W2 showing https://bar.com/b.html which has an iframe containing https://foo.com/a.html which uses the same shared worker spawned above.
* I have window W1 open and showing in my screen.
* I have window W2 open and next to W1, but the focus is on the top-level frame, which is not same origin with the iframe.

This means that the iframe in W2 can potentially have access to updates and readings from the shared worker that would not be available if it had been using the API directly rather than via a shared worker.

I don't have a lot of experience with shared workers to know if this is common or OK, but it'd be good to mention this in https://w3c.github.io/compute-pressure/#security-and-privacy-considerations (perhaps in https://w3c.github.io/compute-pressure/#same-origin-restriction but maybe elsewhere).

Reply:
I think that is OK. We want to enable usage in iframes and that additionally depends on the embedder allowing the feature policy for the frame. The only way the W2 would have access is if it collaborates with the iframe to get that info

Reply:
This brings up an interesting topic in that the permissions policy API does not apply to workers at the moment IIRC, so this means that not-same origin iframe can use the Compute Pressure API in a worker/shared worker but would not be able to do so if it was using it directly from its code, right?

Also, my concern wasn't about W2 specifically but rather if it's OK that the iframe in W2 would receive updates from its worker.

Reply:
Yes, I think it is OK. We have permissions policy the embedder can control and it was one of the core use-cases we had from Zoom: it's going to be used from iframes and they want to share the worker across instances.

Also we are not showing a permissions prompt for the feature which is what Google wanted. Users won't understand what it means to use computer pressure

Reply:
But yeah, it would also be weird if the shared worker cannot get data because one embedder didn't allow it.

I really don't know what to do in this situation.

Reply:
I guess we could check them all and the app code would have to make sure to check policy before deciding to use a shared worker, and then document that in an example

Reply:
Isn't it dangerous if it's up to the app as well?

My concern here is that apps could work around the privacy checks being defined for windows -- that is, if a page uses the API directly and the privacy checks would fail under certain circumstances, all it takes is using the API from a shared worker and having another tab/window loading the same shared worker and passing the privacy checks for the first page to bypass the mitigations.

Reply:
Yeah so let's require that all in 'owner set' have appropriate permissions policy